ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+                pMMAdmDB->pMMBDXP, nIField, nNewWidth,
+                pMMAdmDB->pMMBDXP->pField[nIField].DecimalsIfFloat,
+                (MM_BYTE)MM_NOU_N_DECIMALS_NO_APLICA))
+            return 1;
+
+        // The record on course also has to change its size.
+        if ((GUInt64)pMMAdmDB->pMMBDXP->BytesPerRecord + 1 >=
+            pMMAdmDB->nNumRecordOnCourse)
+        {
+            if (nullptr == (pMMAdmDB->szRecordOnCourse = realloc_function(
+                                pMMAdmDB->szRecordOnCourse,
+                                (size_t)pMMAdmDB->pMMBDXP->BytesPerRecord + 1)))
+            {
+                MMCPLError(CE_Failure, CPLE_OutOfMemory,
+                           "Memory error in MiraMon "
+                           "driver (MMTestAndFixValueToRecordDBXP())");
+                return 1;
+            }
+        }
+
+        // File has changed it's size, so it has to be updated
+        // at the Flush tool
+        fseek_function(pMMAdmDB->pFExtDBF, 0, SEEK_END);
+        pMMAdmDB->FlushRecList.OffsetWhereToFlush =
+            ftell_function(pMMAdmDB->pFExtDBF);
+    }
+    return 0;
+}
+
+int MMWriteValueToRecordDBXP(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                             char *registre, const struct MM_FIELD *camp,
+                             const void *valor, MM_BOOLEAN is_64)
+{
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (!camp)
+        return 0;
+
+    if (MMResizeStringToOperateIfNeeded(hMiraMonLayer,
+                                        camp->BytesPerField + 10))
+        return 1;
+
+    if (!valor)
+        *hMiraMonLayer->szStringToOperate = '\0';
+    else
+    {
+        if (camp->FieldType == 'N')
+        {
+            if (!is_64)
+            {
+                sprintf(hMiraMonLayer->szStringToOperate, "%*.*f",
+                        camp->BytesPerField, camp->DecimalsIfFloat,
+                        *(const double *)valor);
+            }
+            else
+            {
+                sprintf(hMiraMonLayer->szStringToOperate, "%*lld",
+                        camp->BytesPerField, *(const GInt64 *)valor);
+            }
+        }
+        else
+        {
+            sprintf(hMiraMonLayer->szStringToOperate, "%-*s",
+                    camp->BytesPerField, (const char *)valor);
+        }
+    }
+
+    memcpy(registre + camp->AcumulatedBytes, hMiraMonLayer->szStringToOperate,
+           camp->BytesPerField);
+    return 0;
+}
+
+// Gets the n-th value of the format (number_of_values:val1,val2,...,valN)
+char *MMGetNFieldValue(const char *pszStringList, GUInt32 nIRecord)
+{
+    char *p, *q;
+    GUInt32 nNValues, nIValues;
+    char *pszAux;
+
+    if (!pszStringList)
+        return nullptr;
+
+    pszAux = strdup_function(pszStringList);
+    p = strstr(pszAux, "(");
+    if (!p)
+        return nullptr;
+    p++;
+    if (!p)
+        return nullptr;
+    q = strstr(p, ":");
+    p[(ptrdiff_t)q - (ptrdiff_t)p] = '\0';
```

Comment:
couldn't that be just written?
```suggestion
   *q = '\0';
```

Reply:
Yes. Thanks